core/src/main/java/org/apache/carbondata/hadoop/CarbonInputSplit.java

```diff
@@ -439,8 +434,19 @@ public Segment getSegment() {
       out.writeShort(blockletId);
     }
     out.writeBoolean(isLegacyStore);
+    writeDeleteDeltaFile(out);
+  }
+
+  private void writeDeleteDeltaFile(DataOutput out) throws IOException {
```

Comment:
In case of index server ...index executor will not write delete delta information, so we need to add a check if it's a index server case do not write delete delta file, otherwise reading will not work

Reply:
done